### PR TITLE
Migrate to postgres and add more quips

### PR DESCRIPTION
**Pull request by AhmedNSidd**
This pull request migrates our bot to a POSGRES db instead of the json files we were using previously. 

Overview: 
```
- JSON files were deleted (db/notify.json & db/status.json)
- Created general/utils.py which has a SQL array creator which basically just constructs an array in SQL syntax to be used for SQL queries.
- Massive changes to handler/notify.py and handler/status.py which is basically just using psycopg2 to communicate with our posgres database. 
- Added  psycopg2 to our requirements
- Added 2 more quips
```

Room for improvement: 
```
- Probably should add more inline comments, and the comments I did add are pretty inconsistently placed function-to-function as well.
- Should probably also document the database schemas I made on the db side
```

This code is already in production for our bot.

### CODE REVIEW

File: handlers/basic.py [hunk 1]
```diff
@@ -23,10 +23,9 @@ def quip(update, context):
     update.message.reply_text(random.choice(values.quips))
 
 def get_group_id(update, context):
-    """Sends a quip everytime the command /quip is issued."""
+    """Replies with a group's chat ID"""
```
Comment (just-a-jazz) [inline handlers/basic.py:26]: Minor: "Returns" would make more sense here than "Replies"

File: handlers/notify.py [hunk 2]
```diff
@@ -1,135 +1,135 @@
...
+            return
+        group_id = str(update.effective_chat.id)
+        with conn.cursor() as cursor:
+            new_notify_msg = " ".join(context.args)
+            cursor.execute('SELECT 1 FROM notifications WHERE id = {}'.format(group_id))
+            group_exists = cursor.fetchone()
+            if group_exists:
+                cursor.execute("UPDATE notifications SET message = '{}' WHERE id = {}".format(new_notify_msg, group_id))
+            else:
+                cursor.execute("INSERT INTO notifications (id, message) VALUES({}, '{}')".format(group_id, new_notify_msg))
```
Comment (just-a-jazz) [inline handlers/notify.py:22]: I recommend abstracting out this database code into its own class. You can have the url variable there, and different methods for fetching, inserting and updating rows. Then from this code you'd call the DB class methods. Also any time where you have constant strings like `SELECT 1 FROM notifications WHERE id = {}`, I recommend putting them into a static variable in the DB class. 

You can find more info here:
https://betterprogramming.pub/singleton-in-python-5eaa66618e3d
https://stackoverflow.com/a/60284828

File: handlers/notify.py [hunk 3]
```diff
@@ -1,135 +1,135 @@
...
+
+                # Build up a query to insert new users plus any pre-existing ones
+                query = f"UPDATE notifications SET users = {create_sql_array(list(users))} WHERE id = {group_id}"
+                cursor.execute(query)
+            else:
+                # Build up a query to insert new users as a new record in the table
+                query = f"INSERT INTO notifications (id, users) VALUES({group_id}, {create_sql_array(context.args)})"
+                cursor.execute(query)
+
+        update.message.reply_text("Done! The new user(s) have been added.")
```
Comment (just-a-jazz) [inline handlers/notify.py:50]: Once you have a DB class, this code can be modified to use that class. The same goes for every other method in this PR that makes use of DB code.

File: handlers/status.py [hunk 4]
```diff
@@ -1,97 +1,107 @@
...
+            curr_record = cursor.fetchone()
+            if not curr_record or not curr_record[0]:
+                update.message.reply_text("No users have been set to show status of.")
+            else:
+                for x in range(len(curr_record[0])):
+                    try:
+                        user = XboxApi.get_player(curr_record[1][x], curr_record[0][x])
+                        players.append(str(user))
+                    except requests.HTTPError:
+                        update.message.reply_text("Something went wrong with the API.")
```
Comment (just-a-jazz) [inline handlers/status.py:103]: You can simplify this for loop by using zip. For example:

```
for (name, id) in zip(curr_record):
    try:
        user = XboxApi.get_player(id, name)
        players.append(str(user))
    except requests.HTTPError:
        update.message.reply_text("Something went wrong with the API.")
```

File: handlers/status.py [hunk 5]
```diff
@@ -1,97 +1,107 @@
...
-                    user = XboxApi.get_player(status_content[group_id][name], name)
-                    players.append(str(user))
-                except requests.HTTPError:
-                    update.message.reply_text("Something went wrong with the API.")
-            players.sort()
+    with psycopg2.connect(dbname=url.path[1:],user=url.username,password=url.password,host=url.hostname,port=url.port) as conn:
+        players = []
+        group_id = str(update.effective_chat.id)
+        with conn.cursor() as cursor:
+            cursor.execute('SELECT name, xuid FROM status WHERE id = {}'.format(group_id))
```
Comment (just-a-jazz) [inline handlers/status.py:93]: If you change the query to the following:

`SELECT xuid, name FROM status WHERE id = {}`

then the for loop below will be able to use (id, name) and thus be consistent with the ordering introduced in the interface for XpoxApi's get_player method.

File: handlers/basic.py (hunk 1 above)
Comment (AhmedNSidd) [reply handlers/basic.py:26]: Would it? It's not really returning anything code-wise. It's replying to a user's command in the telegram chat. Maybe making it more descriptive, something link: `Replies to the user's message with the group's chat ID` would be better?

**Pull request merged**